### PR TITLE
make Compressor in Audio Output optional

### PR DESCRIPTION
**Pull request by friedolino78**
in this branch the StereoCompressor in the audio output will be changed to be optional an per default off to keep compatibility.

### CODE REVIEW

**Comment by friedolino78** [comment]
local test results are fine. maybe a problem with a different compiler?

File: src/Nio/OutMgr.cpp [hunk 1]
```diff
@@ -125,6 +125,17 @@ string OutMgr::getSink() const
     return "ERROR";
 }
 
+void OutMgr::setAudioCompressor(bool isEnabled)
+{
+    printf("OutMgr - setting isOutputCompressionEnabled to: %d\n", isEnabled);
```
Comment (fundamental) [inline src/Nio/OutMgr.cpp:130]: Good for development, but this should be removed for the final merging.

File: src/Nio/OutMgr.h [hunk 2]
```diff
@@ -49,8 +49,11 @@ class OutMgr
         std::string getDriver() const;
 
         bool setSink(std::string name);
-
+
```
Comment (fundamental) [inline src/Nio/OutMgr.h:52]: Nit pick, but extra whitespace.

**Comment by fundamental** [comment]
Looks like the compiler issue may likely be related to the stub implementation of Nio for when plugins are built. See the end of the src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp for the one stub impl I could recall.

File: src/Nio/OutMgr.cpp (hunk 1 above)
Comment (friedolino78) [reply src/Nio/OutMgr.cpp]: ok

**Comment by friedolino78** [comment]
> Looks like the compiler issue may likely be related to the stub implementation of Nio for when plugins are built. See the end of the src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp for the one stub impl I could recall.

it helped. i did it in also in DSSI. somehow they differ a little bit (waveEnd) maybe you have another look whether everything is correct.

**Comment by fundamental** [comment]
It should be good if everything is building. One modification that I thought about today is saving the user's choice in the global settings xml rather than forcing them to enter in their choice every start. Opinions?

**Comment by friedolino78** [comment]
> It should be good if everything is building. One modification that I thought about today is saving the user's choice in the global settings xml rather than forcing them to enter in their choice every start. Opinions?

it's a good idea.

**Comment by fundamental** [comment]
Thanks for the fix :+1:

**Pull request merged**